pkg/api/types.go

```diff
@@ -1368,6 +1368,13 @@ type NodeSpec struct {
 	Unschedulable bool `json:"unschedulable,omitempty"`
 }
 
+type Heartbeat struct {
+	// Time when this heartbeat was generated.
```

Comment:
I still think it's unnecessary, as I don't think that heartbeat would be something that "represent a persistent entity" or anything that defines "object entity". I might have missed something, as it's a long read.